src/activity_handlers.h

```diff
@@ -1,36 +1,86 @@
 #ifndef ACTIVITY_HANDLERS_H
 #define ACTIVITY_HANDLERS_H
 
-class player_activity;
+#include "player_activity.h"
+
 class player;
 
+// activity_item_handling.cpp
+void activity_on_turn_drop();
+void activity_on_turn_move_items();
+void activity_on_turn_pickup();
+void activity_on_turn_stash();
+
+// advanced_inv.cpp
+void advanced_inv();
+
+// veh_interact.cpp
+void complete_vehicle();
+
 namespace activity_handlers
 {
+
+/** foo_do_turn functions: */
 void burrow_do_turn( player_activity *act, player *p );
+void fill_liquid_do_turn( player_activity *act, player *p );
+void pickaxe_do_turn( player_activity *act, player *p );
+void drop_do_turn( player_activity *act, player *p );
+void stash_do_turn( player_activity *act, player *p );
+void pulp_do_turn( player_activity *act, player *p );
+void game_do_turn( player_activity *act, player *p );
+void start_fire_do_turn( player_activity *act, player *p );
+void vibe_do_turn( player_activity *act, player *p );
+void oxytorch_do_turn( player_activity *act, player *p );
+void aim_do_turn( player_activity *act, player *p );
+void pickup_do_turn( player_activity *act, player *p );
+void move_items_do_turn( player_activity *act, player *p );
+void adv_inventory_do_turn( player_activity *act, player *p );
+void armor_layers_do_turn( player_activity *act, player *p );
+void atm_do_turn( player_activity *act, player *p );
+void cracking_do_turn( player_activity *act, player *p );
+void repair_item_do_turn( player_activity *act, player *p );
+void butcher_do_turn( player_activity *act, player *p );
+
+static const std::map< string_id<activity_type>, std::function<void( player_activity *, player *)> > do_turn_functions =
+{
+    { activity_id( "ACT_BURROW" ), burrow_do_turn },
+    { activity_id( "ACT_FILL_LIQUID" ), fill_liquid_do_turn },
+    { activity_id( "ACT_PICKAXE" ), pickaxe_do_turn },
+    { activity_id( "ACT_DROP" ), drop_do_turn },
+    { activity_id( "ACT_STASH" ), stash_do_turn },
+    { activity_id( "ACT_PULP" ), pulp_do_turn },
+    { activity_id( "ACT_GAME" ), game_do_turn },
+    { activity_id( "ACT_START_FIRE" ), start_fire_do_turn },
+    { activity_id( "ACT_VIBE" ), vibe_do_turn },
+    { activity_id( "ACT_OXYTORCH" ), oxytorch_do_turn },
+    { activity_id( "ACT_AIM" ), aim_do_turn },
+    { activity_id( "ACT_PICKUP" ), pickup_do_turn },
+    { activity_id( "ACT_MOVE_ITEMS" ), move_items_do_turn },
+    { activity_id( "ACT_ADV_INVENTORY" ), adv_inventory_do_turn },
+    { activity_id( "ACT_ARMOR_LAYERS" ), armor_layers_do_turn },
+    { activity_id( "ACT_ATM" ), atm_do_turn },
+    { activity_id( "ACT_CRACKING" ), cracking_do_turn },
+    { activity_id( "ACT_REPAIR_ITEM" ), repair_item_do_turn },
+    { activity_id( "ACT_BUTCHER" ), butcher_do_turn }
+};
```

Comment:
If possible, it would be better to move as much of it as possible into the `.cpp` file.